### PR TITLE
add volume_multiplier and create bags of holding for magiclysm

### PR DESCRIPTION
**Pull request by KorGgenT**
#### Summary
```SUMMARY: Mods "[Magiclysm] add bags of holding"```

#### Purpose of change
what fantasy RPG with inventory management would be complete without bags of holding?

#### Describe the solution
created a new pocket member called volume_multiplier that makes it so when you put an item into a pocket with this multiplier, the nonrigid pocket wille xpand at a smaller rate than the volume you place into it. Additionally, added a couple tiers of bags of holding and added tier 1 and 2 to item groups. tier 3 will be in some other PR, when i or someone makes an NPC shop.

#### Testing
Spawned in the bag, put some items in it, and made sure the bag's stats made sense - it was smaller than the sum of its parts, and it weighs less too!

#### Additional context
I'd really like an npc magic shop that you can find where you can buy crazy stuff like this, and that has backup in the form of magic turrets that should heavily discourage murdering the NPC for the sweet loot.

### CODE REVIEW

**Comment by KorGgenT** [comment]
image

File: data/mods/Magiclysm/itemgroups/itemgroups.json [hunk 1]
```diff
@@ -180,7 +180,8 @@
           { "group": "enchanted_rings_common", "prob": 20 },
           { "group": "enchanted_rings_uncommon", "prob": 5 },
           { "group": "enchanted_combat_items", "prob": 10 },
-          { "group": "enchanted_belts", "prob": 2 }
+          { "group": "enchanted_belts", "prob": 2 },
+          { "group": "enchanted_mish", "prob": 1 }
```
Comment (ave-) [inline data/mods/Magiclysm/itemgroups/itemgroups.json:184]:
```suggestion
          { "group": "enchanted_misc", "prob": 1 }
```

**Comment by Aphegis** [comment]
Loved it

**Comment by dissociativity** [comment]
could you perhaps add the infrastructure to make compression bags for soft items like cotton and wool clothes in the vanilla game for this?
They work quite well for sleeping bags/blankets especially, or maybe your cloak

File: src/item_pocket.cpp [hunk 2]
```diff
@@ -667,6 +665,12 @@ void item_pocket::general_info( std::vector<iteminfo> &info, int pocket_number,
                            string_format( _( "Items in this pocket weigh <neutral>%.0f%%</neutral> their original weight." ),
                                           data->weight_multiplier * 100 ) );
     }
+    if( data->volume_multiplier != 1.0f ) {
+        info.emplace_back( "DESCRIPTION",
+                           string_format(
+                               _( "This pocket expands at <neutral>%.0f%%</neutral> of the rate of volume of items inside." ),
+                               data->weight_multiplier * 100 ) );
```
Comment (Cenness) [inline src/item_pocket.cpp:672]: Did you mean volume_multiplier here?

Comment (KorGgenT) [reply src/item_pocket.cpp:672]: i sure did!

Comment (ave-) [reply src/item_pocket.cpp:672]: Sorry, I've missed this during my review.

**Pull request merged**